plugins/single_plugins/fisheye.cpp

```diff
@@ -0,0 +1,201 @@
+                        current_zoom = target_zoom;
+
+                        auto current = duration.progress();
+                        duration.start(current, target_zoom);
+
+                        if (!hook_set)
+                        {
+                            hook_set = true;
+                            output->render->add_post(&hook);
+                        }
+                    }
+            };
+
+            auto vs = OpenGL::compile_shader(vertex_shader, GL_VERTEX_SHADER);
+            auto fs = OpenGL::compile_shader(fragment_shader, GL_FRAGMENT_SHADER);
+
+            program = GL_CALL(glCreateProgram());
+            GL_CALL(glAttachShader(program, vs));
+            GL_CALL(glAttachShader(program, fs));
+            GL_CALL(glLinkProgram(program));
+
+            posID = GL_CALL(glGetAttribLocation(program, "position"));
+            mouseID  = GL_CALL(glGetUniformLocation(program, "u_mouse"));
+            resID  = GL_CALL(glGetUniformLocation(program, "u_resolution"));
+            radiusID  = GL_CALL(glGetUniformLocation(program, "u_radius"));
+            zoomID  = GL_CALL(glGetUniformLocation(program, "u_zoom"));
+
+            duration = wf_duration(new_static_option("700"));
+            duration.start(1, 1); // so that the first value we get is correct
+
+            output->add_key(toggle_key, &toggle_cb);
+        }
+
+        void render(uint32_t fb, uint32_t tex, uint32_t target)
+        {
+            GetTuple(x, y, output->get_cursor_position());
+            GL_CALL(glUseProgram(program));
+            GL_CALL(glBindTexture(GL_TEXTURE_2D, tex));
+            GL_CALL(glActiveTexture(GL_TEXTURE0));
+
+            static const float vertexData[] = {
+                -1.0f, -1.0f,
+                 1.0f, -1.0f,
+                 1.0f,  1.0f,
+                -1.0f,  1.0f
+            };
+
+            radius = radius_opt->as_double();
+            target_zoom = zoom_opt->as_double();
+
+            current_zoom = duration.progress();
+
+            glUniform2f(mouseID, x, y);
+            glUniform2f(resID, output->handle->width, output->handle->height);
+            glUniform1f(radiusID, radius);
+            glUniform1f(zoomID, current_zoom);
+
+            GL_CALL(glVertexAttribPointer(posID, 2, GL_FLOAT, GL_FALSE, 0, vertexData));
+            GL_CALL(glEnableVertexAttribArray(posID));
+
+            GL_CALL(glDisable(GL_BLEND));
+            GL_CALL(glBindFramebuffer(GL_DRAW_FRAMEBUFFER, target));
+            GL_CALL(glDrawArrays (GL_TRIANGLE_FAN, 0, 4));
+            GL_CALL(glBindFramebuffer(GL_FRAMEBUFFER, 0));
+
+            GL_CALL(glDisableVertexAttribArray(posID));
+            GL_CALL(glBindTexture(GL_TEXTURE_2D, 0));
+
+            if (!duration.running() && target_zoom == 0)
+            {
+                output->render->rem_post(&hook);
+                hook_set = false;
+            }
+
+            if (active)
+                duration.start(current_zoom, target_zoom);
```

Comment:
I can't get why you start at every frame ..

Reply:
This is so changing target_zoom via wcm works.